src/utils/restApi/api.tsx

```diff
@@ -46,70 +49,79 @@ const getDesired = (data: DesiredType) => {
 };
 
 export const getContracts = async (
-  client: RestAPI,
+  client: RestClient,
   setData: Dispatch<SetStateAction<ITableData[] | null>>,
   setError: Dispatch<SetStateAction<string | null>>,
+  searchQuery: string,
 ) => {
   try {
     const range = contractsRange(`contractId;limit 10;offset 0;order desc`);
 
-    const allContracts = await client.getContracts({
-      tags: [`${env.NEXT_PUBLIC_DAPP_ID}`],
-      range,
-    });
-
-    const parsedContracts = contractSchema.safeParse(allContracts);
+    const tags =
+      searchQuery !== ""
+        ? [env.NEXT_PUBLIC_DAPP_ID + SWAP_TAG + searchQuery.toLocaleLowerCase()]
```

Comment:
This string interpolation (that I also mentioned in the other file) seems a good candidate to be extracted as a function so that we make sure we use the same tag.

Reply:
Great! I'll change it